service/javadsl/api/src/main/scala/com/lightbend/lagom/internal/api/ServiceReader.scala

```diff
@@ -37,9 +37,14 @@ object ServiceReader {
   }
 
   def readServiceDescriptor(classLoader: ClassLoader, serviceInterface: Class[_ <: Service]): Descriptor = {
-    val invocationHandler = new ServiceInvocationHandler(classLoader, serviceInterface)
-    val serviceStub = JProxy.newProxyInstance(classLoader, Array(serviceInterface), invocationHandler).asInstanceOf[Service]
-    serviceStub.descriptor()
+    if (Modifier.isPublic(serviceInterface.getModifiers)) {
+      val invocationHandler = new ServiceInvocationHandler(classLoader, serviceInterface)
+      val serviceStub = JProxy.newProxyInstance(classLoader, Array(serviceInterface), invocationHandler).asInstanceOf[Service]
+      serviceStub.descriptor()
+    } else {
+      // If the default descriptor method is implemented in a non-public interface, throw an exception.
+      throw new IllegalArgumentException("Service API must be described in a public interface.")
+    }
```

Comment:
Just a minor style comment, feel free to ignore if you disagree: I like to write these kinds of checks in a guard clause style, with the exception raised on the first or second line if the precondition isn't met, and the rest of the normal execution un-nested from the conditional. It keeps the check and the consequence next to each other in the code, and highlights the "expected" code path.

Scala has the [`Predef.require`](http://www.scala-lang.org/api/2.11.8/index.html#scala.Predef$@require(requirement:Boolean,message:=>Any):Unit) convenience function for this, which I like because it reduces the noise around the precondition check and makes the code more self-documenting.

```scala
def readServiceDescriptor(classLoader: ClassLoader, serviceInterface: Class[_ <: Service]): Descriptor = {
  require(Modifier.isPublic(serviceInterface.getModifiers), "Service API must be described in a public interface.")

  val invocationHandler = new ServiceInvocationHandler(classLoader, serviceInterface)
  val serviceStub = JProxy.newProxyInstance(classLoader, Array(serviceInterface), invocationHandler).asInstanceOf[Service]
  serviceStub.descriptor()
}
```